src/test/kotlin/com/zepben/auth/client/ZepbenTokenFetcherTest.kt

```diff
@@ -299,7 +299,29 @@ internal class ZepbenTokenFetcherTest {
             bodyPublishers.verify { BodyPublishers.ofString(matches("audience=test_audience&refresh_token=test_refresh_token")) }
             assertThat(token, equalTo("Bearer $TOKEN"))
         }
+    }
+
+    @Test
+    fun testFetchEntraTokenSuccessfulUsingRefresh() {
+        doReturn(StatusCode.OK.code).`when`(response).statusCode()
+        doReturn(
+            "{\"access_token\":\"$TOKEN\", \"refresh_token\": \"test_refresh_token\", \"token_type\":\"Bearer\"}"
+        ).`when`(response).body()
 
+        mockStatic(BodyPublishers::class.java, CALLS_REAL_METHODS).use { bodyPublishers ->
+            val tokenFetcher = ZepbenTokenFetcher(
+                audience = "test_audience",
+                tokenEndpoint = "https://testissuer.com.au",
+                authMethod = AuthMethod.ENTRAID,
+                client = client,
+                refreshToken = "test_refresh_token"
+            )
+            verify(client, never()).send(any(), any<HttpResponse.BodyHandler<String>>())
+            val token = tokenFetcher.fetchToken()
+            verify(client).send(any(), any<HttpResponse.BodyHandler<String>>())
+            bodyPublishers.verify { BodyPublishers.ofString(matches("scope=test_audience/.default&refresh_token=test_refresh_token")) }
+            assertThat(token, equalTo("Bearer $TOKEN"))
+        }
```

Comment:
This is nearly identical to Auth0 test, if we can make them share code somehow, that would be great. I'm just not familiar enough with mocks to make it work.